src/stats/query_result_stats.rs

```diff
@@ -0,0 +1,68 @@
+use crate::errors::Error;
+use crate::query::Query;
+use lru::LruCache;
+use std::num::NonZeroUsize;
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Key {
+    pub fingerprint: u64,
+    pub normalized: String,
+    pub result_hash: Vec<u8>,
+}
+
+#[derive(Debug)]
+pub struct Value {
+    pub count: usize,
+    pub first_seen: chrono::DateTime<chrono::Utc>,
+    pub last_seen: chrono::DateTime<chrono::Utc>,
+}
+
+impl Value {
+    pub fn duration(&self) -> chrono::Duration {
+        self.last_seen - self.first_seen
+    }
+}
+
+#[derive(Debug)]
+pub struct QueryResultStats {
+    pub statistics: LruCache<Key, Value>,
+}
+
+impl Default for QueryResultStats {
+    fn default() -> Self {
+        QueryResultStats {
+            statistics: LruCache::new(NonZeroUsize::new(1000).unwrap()),
+        }
+    }
+}
+
+impl QueryResultStats {
+    pub(crate) fn new() -> QueryResultStats {
+        QueryResultStats {
+            statistics: LruCache::new(NonZeroUsize::new(1000).unwrap()),
```

Comment:
Is it worth making this `1000` configurable? At least, it might be worth creating a named `const` for it.

Reply:
yes, I had initially made a config option but reverted it to simplify it locally